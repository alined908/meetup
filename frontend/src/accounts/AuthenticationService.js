class AuthenticationService{

    registerSuccessfulLogin(token){
        localStorage.setItem('token', token);
        console.log("register successfull login");
    }

    logout(){
        localStorage.removeItem('token');
    }

    isUserLoggedIn(){
        let user = localStorage.getItem('token')
        if (user === null){
            return false
        }
        return true
    }
}

export default new AuthenticationService()