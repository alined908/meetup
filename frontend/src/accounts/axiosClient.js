import axios from "axios";

export const axiosClient = axios.create({
  baseURL:
    process.env.NODE_ENV === "production"
      ? `${process.env.REACT_APP_PROD_BASE_URL}`
      : `${process.env.REACT_APP_DEV_BASE_URL}`,
});
