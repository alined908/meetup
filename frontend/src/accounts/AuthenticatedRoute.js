import React, { Component } from "react";
import AuthenticationService from "./AuthenticationService";
import { Route, Redirect } from "react-router-dom";

class AuthenticatedRoute extends Component {
  render() {
    if (AuthenticationService.isUserLoggedIn()) {
      return <Route {...this.props} />;
    } else {
      return (
        <Redirect
          to={{
            pathname: "/login/",
            state: {
              from: this.props.location.pathname,
            },
          }}
        />
      );
    }
  }
}

export default AuthenticatedRoute;
