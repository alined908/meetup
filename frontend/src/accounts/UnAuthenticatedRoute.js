import React, { Component } from "react";
import AuthenticationService from "./AuthenticationService";
import { Route, Redirect } from "react-router-dom";

class UnAuthenticatedRoute extends Component {
  render() {
    if (!AuthenticationService.isUserLoggedIn()) {
      return (
        <Route {...this.props} other={this.props.location.pathname}></Route>
      );
    } else {
      return <Redirect to="/feed"></Redirect>;
    }
  }
}

export default UnAuthenticatedRoute;
