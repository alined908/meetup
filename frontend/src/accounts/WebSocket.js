import AuthenticationService from "./AuthenticationService";

export default class WebSocketService {
  callbacks = {};

  constructor() {
    this.socketRef = null;
    this.waitForSocketConnection = this.waitForSocketConnection.bind(this);
    this.disconnect = this.disconnect.bind(this);
    this.connect = this.connect.bind(this);
  }

  getCallbacks() {
    return this.callbacks;
  }

  connect(path, token) {
    const baseURL =
      process.env.NODE_ENV === "production"
        ? process.env.REACT_APP_PROD_BASE_URL
        : process.env.REACT_APP_DEV_BASE_URL;
    const urlWithToken = baseURL;
    var url = new URL(path, urlWithToken);
    url.protocol =
      url.protocol === "http:"
        ? url.protocol.replace("http", "ws")
        : url.protocol.replace("https", "wss");
    if (token === null) {
      return;
    }
    this.socketRef = new WebSocket(url.href + `?token=${token}`);

    this.socketRef.onmessage = (e) => {
      console.log("Step 2 - Channel sends message from backend");
      this.socketNewMessage(e.data);
    };

    this.socketRef.onopen = () => {
      console.log("WebSocket open");
    };

    this.socketRef.onerror = (e) => {
      console.log(e.message);
    };

    this.socketRef.onclose = (e) => {
      console.log(e);
      //Token not authenticated
      console.log(e.code)
      const access = AuthenticationService.retrieveToken();
      setTimeout(() => this.connect(path, access), 500);
    //   if (e.code === 4000) {
    //     console.log("this reached")
    //     const access = AuthenticationService.retrieveToken();
    //     setTimeout(() => this.connect(path, access), 3000);
    //   } else {
    //     console.log("that reached")
    //     setTimeout(() => this.connect(path, token), 3000);
    //   }
    };
  }

  disconnect() {
    console.log("disconnect called");
    if (this.socketRef) {
      this.socketRef.onclose = function () {};
      this.socketRef.close();
    }
  }

  socketNewMessage(data) {
    const parsedData = JSON.parse(data);
    const command = parsedData.command;
    if (Object.keys(this.callbacks).length === 0) {
      return;
    }
    console.log("Step 3 - " + command + " command");

    //Convert all ifs to this.callbacks[command](parsedData)
    if (command === "fetch_messages") {
      this.callbacks[command](parsedData.messages);
    } else if (command === "new_message") {
      this.callbacks[command](parsedData.message);
    } else {
      console.log(parsedData);
      this.callbacks[command](parsedData);
    }
  }

  sendMessage(data) {
    console.log("Step 1 - Send Message to Channel");
    try {
      console.log({ ...data });
      console.log(this.socketRef.readyState);
      console.log(this.socketRef.url);
      this.socketRef.send(JSON.stringify({ ...data }));
    } catch (err) {
      console.log(err.message);
    }
  }

  ref() {
      return this.socketRef
  }

  exists() {
    return this.socketRef !== null;
  }

  state() {
    return this.socketRef.readyState;
  }

  uri() {
    return this.socketRef.uri;
  }

  waitForSocketConnection(callback) {
    const socket = this.socketRef;
    const recursion = this.waitForSocketConnection;
    setTimeout(function () {
      if (socket.readyState === 1) {
        console.log("Connection is made");
        if (callback != null) {
          callback();
        }
        return;
      } else {
        // console.log("Wait for connection..");
        recursion(callback);
      }
    }, 2000);
  }
  //Chat commands
  fetchMessages(uri) {
    console.log("Websocket - fetchMessages");
    this.sendMessage({ command: "fetch_messages", uri: uri });
  }

  newChatMessage(message) {
    console.log("Websocket - newChatMessage");
    this.sendMessage({
      command: "new_message",
      from: message.from,
      text: message.text,
      room: message.room,
    });
  }

  newMeetupEvent(data) {
    console.log("Websocket - newEvent");
    this.sendMessage({ command: "new_event", data: data });
  }

  reloadMeetupEvent(data) {
    console.log("Websocket - reloadEvent");
    this.sendMessage({ command: "reload_event", data: data });
  }

  decideMeetupEvent(data) {
    console.log("Websocket - decideEvent");
    this.sendMessage({ command: "decide_event", data: data });
  }

  deleteMeetupEvent(data) {
    console.log("Websocket - delete meetup event");
    this.sendMessage({ command: "delete_event", data: data });
  }

  redecideMeetupEvent(data) {
    console.log("Websocket - redecideEvent");
    this.sendMessage({ command: "redecide_event", data: data });
  }

  voteMeetupEvent(data) {
    console.log("Websocket - voteEvent");
    this.sendMessage({ command: "vote_event", data: data });
  }

  addEventOption(data) {
    console.log("Websocket - addMeetupEventOption");
    this.sendMessage({ command: "new_option", data: data });
  }

  deleteEventOption(data) {
    console.log("Websocket - deleteMeetupEventOption");
    this.sendMessage({ command: "delete_option", data: data });
  }

  //Invite Commands
  fetchInvites(user) {
    console.log("Websocket - fetchInvites");
    this.sendMessage({ command: "fetch_invites", user: user });
  }

  newInvite(user) {
    console.log("Websocket - newInvite");
    this.sendMessage({ command: "new_invite", user: user });
  }

  fetchNotifications(data) {
    console.log("Websocket - fetchNotifications");
    this.sendMessage({ command: "fetch_notifications", data: data });
  }

  readNotifications(user, type) {
    console.log("Websocket - readNotifications");
    this.sendMessage({ command: "read_notifs", user: user, type: type });
  }

  addChatCallbacks(messagesCallback, newMessageCallback) {
    this.callbacks["fetch_messages"] = messagesCallback;
    this.callbacks["new_message"] = newMessageCallback;
  }

  addContactsCallbacks(reloadCallback) {
    this.callbacks["update_room"] = reloadCallback;
  }

  addMeetupCallbacks(
    eventsCallback,
    newEventCallback,
    reloadEventCallback,
    voteEventCallback,
    decideEventCallback,
    deleteEventCallback,
    addMemberCallback,
    deleteMemberCallback,
    newOptionCallback,
    deleteOptionCallback
  ) {
    this.callbacks["fetch_events"] = eventsCallback;
    this.callbacks["new_event"] = newEventCallback;
    this.callbacks["reload_event"] = reloadEventCallback;
    this.callbacks["vote_event"] = voteEventCallback;
    this.callbacks["decide_event"] = decideEventCallback;
    this.callbacks["redecide_event"] = decideEventCallback;
    this.callbacks["delete_event"] = deleteEventCallback;
    this.callbacks["new_member"] = addMemberCallback;
    this.callbacks["delete_member"] = deleteMemberCallback;
    this.callbacks["new_option"] = newOptionCallback;
    this.callbacks["delete_option"] = deleteOptionCallback;
  }

  addNotifCallbacks(addNotifCallback) {
    this.callbacks['create_notif'] = addNotifCallback
  }
}
