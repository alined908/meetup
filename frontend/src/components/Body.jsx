import React, {Component} from 'react';
import {Route, Switch} from 'react-router-dom';
import LogoutComponent from "./auth/LogoutComponent";
import ErrorComponent from "./auth/ErrorComponent";
import RegisterComponent from "./auth/RegisterComponent";
import WelcomeComponent from "./WelcomeComponent";
import ChatComponent from "./chat/ChatComponent"
import MeetupsComponent from "./meetup/MeetupsComponent";
import HomeComponent from "./HomeComponent";
import MeetupPageComponent from "./meetup/MeetupPageComponent"
import FriendsComponent from './invite/FriendsComponent';
import MeetupEventForm from "./meetup/MeetupEventForm"
import AuthenticatedRoute from "../accounts/AuthenticatedRoute";
import LoginComponent from "./auth/LoginComponent";
import Invites from "./invite/Invites"
import MeetupForm from "./meetup/MeetupForm"
import Profile from "./Profile"

class Body extends Component {

    render () {
        return (
            <div className="c">
                <Switch>
                    <Route path="/" exact component={HomeComponent}></Route>
                    <Route path="/login" component={LoginComponent}></Route>
                    <Route path="/register" component={RegisterComponent}></Route>
                    <AuthenticatedRoute path="/logout" component={LogoutComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/welcome" component={WelcomeComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups/:uri/new" exact component={MeetupEventForm}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups/new" exact component={MeetupForm}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups/:uri" component={MeetupPageComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/meetups" exact component={MeetupsComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/chat/:uri" component={ChatComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/chat" exact component={ChatComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/friends" component={FriendsComponent}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/invites" component={Invites}></AuthenticatedRoute>
                    <AuthenticatedRoute path="/profile/:id" component={Profile}></AuthenticatedRoute>
                    <Route component={ErrorComponent}></Route>
                </Switch>
            </div>
        )
    }
}

export default Body