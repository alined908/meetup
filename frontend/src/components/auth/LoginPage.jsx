import React, { Component } from "react";
import { reduxForm, Field } from "redux-form";
import { compose } from "redux";
import { connect } from "react-redux";
import { signin, signinHelper} from "../../actions";
import { Paper, Grid, Fab } from "@material-ui/core";
import FacebookIcon from '@material-ui/icons/Facebook';
import { Link } from "react-router-dom";
import { renderTextField } from "../components";
import {axiosClient} from '../../accounts/axiosClient'
import { ReactComponent as Fan } from "../../assets/svgs/fans.svg";
import styles from "../../styles/form.module.css";
import { GoogleLogin } from 'react-google-login';
import FacebookLogin from 'react-facebook-login'
import { Helmet } from "react-helmet";
import PropTypes from "prop-types";

const validate = (values) => {
  const errors = {};
  if (!values.email) {
    errors.email = "Email is required.";
  }

  if (!values.password) {
    errors.password = "Password is required.";
  }

  return errors;
};

class LoginPage extends Component {
  handleFacebookSocialAuth = async (res) => {
    if (res.error) {
      return;
    }
    console.log(res)
    const response = await axiosClient.post(
      '/auth/facebook/', {accessToken: res.accessToken, email: res.email, name: res.name}
    )
    console.log(response.data)
    signinHelper(response.data, this.props.dispatch, () => this.props.history.push("/meetups"))
  }


  handleGoogleSocialAuth = async (res) => {
    if (res.error){
      return;
    }
    const response = await axiosClient.post(
      '/auth/google/', {tokenId: res.tokenId}
    )
    console.log(response.data)
    signinHelper(response.data, this.props.dispatch, () => this.props.history.push("/meetups"))
  }

  onSubmit = (formProps) => {
    let redirect;
    if (this.props.location.state && this.props.location.state.from) {
      redirect = () => {
        this.props.history.push(this.props.location.state.from);
      };
    } else {
      redirect = () => {
        this.props.history.push("/meetups");
      };
    }
    this.props.signin(formProps, redirect);
  };

  render() {
    const { handleSubmit, submitting, invalid } = this.props;

    return (
      <Paper className={styles.container} elevation={8}>
        <Helmet>
          <meta charSet="utf-8" />
          <title>Login</title>
        </Helmet>
        <div className={styles.left}>
          <Fan className="svg-shadow" height="70%" width="70%" />
        </div>
        <div className={styles.right}>
          <div className={styles.formhead}>
            <span className={styles.header}>Login</span>
          </div>
          <form className={styles.form} onSubmit={handleSubmit(this.onSubmit)}>
            <Grid container spacing={2}>
              <Grid item xs={12}>
                <Field name="email" component={renderTextField} label="Email" />
              </Grid>
              <Grid item xs={12}>
                <Field
                  name="password"
                  type="password"
                  component={renderTextField}
                  label="Password"
                />
              </Grid>
              {this.props.errorMessage && (
                <Grid item xs={12}>
                  <div className={styles.error}>{this.props.errorMessage}</div>
                </Grid>
              )}
            </Grid>
            <div className={styles.fab}>
              <Fab
                color="primary"
                type="submit"
                variant="extended"
                aria-label="login"
                disabled={submitting || invalid}
              >
                Login
              </Fab>
            </div>
          </form>
          <div className={styles.bottom}>
            <div className={styles.socials}>
              <GoogleLogin
                clientId={process.env.REACT_APP_GOOGLE_OAUTH2_KEY}
                buttonText="Continue with Google"
                onSuccess={this.handleGoogleSocialAuth}
                onFailure={this.handleGoogleSocialAuth}
                cookiePolicy={'single_host_origin'}
                className={`${styles.social} ${styles.google}`}
              />
              <FacebookLogin
                autoLoad={false}
                reauthenticate={true}
                appId={process.env.REACT_APP_FACEBOOK_OAUTH2_KEY}
                callback={this.handleFacebookSocialAuth}
                fields="name,email,picture"
                icon={<FacebookIcon/>}
                textButton="Continue with Facebook"
                cssClass={`${styles.social} ${styles.facebook}`}
              />
            </div>
            <div className={styles.action}>
              Don't Have An Account?
              <span className={styles.link}>
                <Link to="/register">Sign Up</Link>
              </span>
            </div>
            <div className={styles.action}>
              Forgot Password?
              <span className={styles.link}>
                <Link to="/password/reset">Reset</Link>
              </span>
            </div>
          </div>
        </div>
      </Paper>
    );
  }
}

LoginPage.propTypes = {
  errorMessage: PropTypes.string,
  signin: PropTypes.func.isRequired,
};

function mapStatetoProps(state) {
  return {
    errorMessage: state.user.loginErrorMessage,
  };
}

const mapDispatchToProps = {
  signin,
};

export default compose(
  connect(mapStatetoProps, mapDispatchToProps),
  reduxForm({
    form: "signin",
    validate
  })
)(LoginPage);

export { LoginPage as UnderlyingLoginPage };
