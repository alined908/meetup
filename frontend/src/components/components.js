//Auth
import ErrorComponent from "./auth/ErrorComponent";
import LoginComponent from "./auth/LoginComponent";
import LogoutComponent from "./auth/LogoutComponent";
import RegisterComponent from "./auth/RegisterComponent";

//Chat
import ChatBarComponent from "./chat/ChatBarComponent"
import ChatComponent from "./chat/ChatComponent"
import ChatMessageComponent from "./chat/ChatMessageComponent"
import ChatWindowComponent from "./chat/ChatWindowComponent"
import ContactComponent from "./chat/ContactComponent"

//invite
import Friend from './invite/Friend';
import FriendsComponent from './invite/FriendsComponent';
import Invite from "./invite/Invite"
import Invites from "./invite/Invites"

//meetup
import Map from "./meetup/Map"
import Meetup from "./meetup/Meetup";
import MeetupCard from "./meetup/MeetupCard";
import MeetupEvent from "./meetup/MeetupEvent";
import MeetupEventForm from "./meetup/MeetupEventForm"
import MeetupForm from "./meetup/MeetupForm";
import MeetupFriend from "./meetup/MeetupFriend";
import MeetupPageComponent from "./meetup/MeetupPageComponent"
import MeetupsComponent from "./meetup/MeetupsComponent";
import Restauraunt from "./meetup/Restauraunt";

//fields
import CategoryAutocomplete from "./fields/CategoryAutocomplete"
import Location from "./fields/Location"
import renderTextField from "./fields/renderTextField"
import renderDatePicker from "./fields/renderDatePicker"
import renderDateSimplePicker from "./fields/renderDateSimplePicker"
import UserAutocomplete from "./fields/UserAutocomplete"

//General
import Body from "./general/Body"
import GlobalMessage from "./general/GlobalMessage"
import HomeComponent from "./general/HomeComponent";
import LiveUpdatingBadge from "./general/LiveUpdatingBadge"
import Navigation from "./general/Navigation"
import Profile from "./general/Profile"
import GroupAvatars from './general/GroupAvatars'
import NotificationWrapper from "./general/NotificationWrapper"
import CalendarComponent from "./general/CalendarComponent"
import CategoriesComponent from "./general/CategoriesComponent"
import CategoryComponent from "./general/CategoryComponent"
import Preferences from "./general/Preferences"
import Preference from "./general/Preference"
import SettingsComponent from "./general/SettingsComponent"

export {
    ErrorComponent, LoginComponent, LogoutComponent, RegisterComponent, 
    ChatBarComponent, ChatComponent, ChatMessageComponent, ChatWindowComponent, ContactComponent,
    Friend, FriendsComponent, Invite, Invites,
    Map, Meetup, MeetupCard, MeetupEvent, MeetupEventForm, MeetupFriend, MeetupForm, MeetupPageComponent, MeetupsComponent, Restauraunt,
    CategoryAutocomplete, UserAutocomplete, Location, renderTextField, renderDatePicker, renderDateSimplePicker,
    Body, GlobalMessage, GroupAvatars, HomeComponent, LiveUpdatingBadge, Navigation, Profile, NotificationWrapper, 
    Preferences, Preference, CalendarComponent, CategoriesComponent, CategoryComponent, SettingsComponent}