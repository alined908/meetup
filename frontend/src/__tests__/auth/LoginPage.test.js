import React from 'react'
import {shallow, mount} from 'enzyme';
import LoginPage, {UnderlyingLoginPage} from "../../components/auth/LoginPage"
import { Provider } from 'react-redux';
import setupStore from "../../setupTests"
import {axiosClient} from "../../accounts/axiosClient";
import MockAdapter from 'axios-mock-adapter';
import { MemoryRouter } from 'react-router';
import {Body} from "../../components/components"
import { createBrowserHistory } from 'history';

describe("LoginPage", () => {
    it ('renders without crashing given props', () => {
        const props = {
            errorMessage: "",
            signin: jest.fn(),
            handleSubmit: jest.fn()
        }
        const wrapper = shallow(<UnderlyingLoginPage {...props}/>)
        expect(wrapper).toMatchSnapshot()
    })

    it('dispatches correct action on submit', () => {
        const props = {
            location: {state: "something"},
            errorMessage: "",
            signin: jest.fn(),
            handleSubmit: jest.fn()
        }
        const mockData = {email: "daniel@gmail.com", password: "password"}

        const wrapper = shallow(<UnderlyingLoginPage {...props}/>)
        wrapper.instance().onSubmit(mockData);
        expect(props.signin.mock.calls.length).toBe(1);
    })

    it ('render errorMessage', () => {
        const props = {
            errorMessage: "Email or password is wrong.",
            signin: jest.fn(),
            handleSubmit: jest.fn()
        }
        const wrapper = shallow(<UnderlyingLoginPage {...props}/>)
        expect(wrapper.find('.error').length).toBe(1)
        expect(wrapper.find('.error').text()).toBe("Email or password is wrong.")
    })
})