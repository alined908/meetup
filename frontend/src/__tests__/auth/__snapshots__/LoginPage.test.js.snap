// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage unit renders without crashing given props 1`] = `
<WithStyles(ForwardRef(Paper))
  className="container"
  elevation={8}
>
  <div
    className="left"
  >
    <Component
      height="70%"
      width="70%"
    />
  </div>
  <div
    className="right"
  >
    <div
      className="formhead"
    >
      <div
        className="icon"
      >
        <Component
          height="100%"
          width="100%"
        />
      </div>
      <span
        className="header"
      >
        Login
      </span>
      <div
        className="icon"
      >
        <Component
          height="100%"
          width="100%"
        />
      </div>
    </div>
    <form
      className="form"
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        spacing={2}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <Field
            component={[Function]}
            label="Email"
            name="email"
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <Field
            component={[Function]}
            label="Password"
            name="password"
            type="password"
          />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <div
        className="fab"
      >
        <WithStyles(ForwardRef(Fab))
          aria-label="login"
          color="primary"
          type="submit"
          variant="extended"
        >
          Login
        </WithStyles(ForwardRef(Fab))>
      </div>
    </form>
    <div
      className="bottom"
    >
      <div
        className="socials"
      >
        <div
          className="social"
        >
          <Component
            height={40}
            width={40}
          />
        </div>
        <div
          className="social"
        >
          <Component
            height={40}
            width={40}
          />
        </div>
        <div
          className="social"
        >
          <Component
            height={40}
            width={40}
          />
        </div>
      </div>
      <div
        className="action"
      >
        Don't Have An Account?
        <span
          className="link"
        >
          <Link
            to="/register"
          >
            Sign Up
          </Link>
        </span>
      </div>
      <div
        className="action"
      >
        Forgot Password?
        <span
          className="link"
        >
          Recover
        </span>
      </div>
    </div>
  </div>
</WithStyles(ForwardRef(Paper))>
`;
